src/support/mtx_rw.c

```diff
@@ -198,37 +163,75 @@ __wt_readlock_spin(WT_SESSION_IMPL *session, WT_RWLOCK *l)
 void
 __wt_readlock(WT_SESSION_IMPL *session, WT_RWLOCK *l)
```

Comment:
I added comments explaining why: the atomic operations in the trylock functions are sufficient, but in the slow path we need a barrier after the thread enters the lock.